server/market.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import "math"
+
+type CommodityType string
+
+const (
+	Tomato    CommodityType = "tomato"
+	Blueberry CommodityType = "blueberry"
+	Corn      CommodityType = "corn"
+	Purple    CommodityType = "purple"
+)
+
+var AllCommodities []CommodityType = []CommodityType{Tomato, Blueberry, Corn, Purple}
+
+type Market struct {
+	Commodities map[CommodityType]*Commodity
+}
+
+func NewMarket() Market {
+	m := Market{Commodities: make(map[CommodityType]*Commodity)}
+
+	for _, c := range AllCommodities {
+		m.Commodities[c] = &Commodity{
+			Name:   string(c),
+			Supply: 100,
+			Value:  100.00,
+			Demand: 100,
+		}
+	}
+
+	return m
+}
+
+func (m *Market) Prices() map[CommodityType]float64 {
+	prices := make(map[CommodityType]float64)
+	for _, c := range AllCommodities {
+		prices[c] = m.Commodities[c].Price()
+	}
+
+	return prices
+}
+
+// Sell returns the unit price of the commodity after it has been sold.
+func (m *Market) Sell(t CommodityType, quantity int64) float64 {
+	c, ok := m.Commodities[t]
+	if !ok {
+		return 0
```

Comment:
Better log error and return nothing

Reply:
I made it return an error. Now, it won't respond with a SaleCompleted message.